File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> anagrams = new HashMap();
        for(String s : strs){
            int[] count = new int[26];
            for(char c : s.toCharArray()){
                count[c - 'a']++;
            }
            String key = Arrays.toString(count);
            if(!anagrams.containsKey(key)){
                anagrams.put(key, new LinkedList());
            }
            System.out.println(key);
            anagrams.get(key).add(s);
        }

        List<List<String>> result = new LinkedList();

        for(Map.Entry<String, List<String>> entry : anagrams.entrySet()){
            result.add(entry.getValue());
        }

        return result;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = [0] * 26
        length = 0
        l = 0
        r = 0
        while r < len(s):
            count[ord(s[r]) - ord('A')] += 1

            while (r - l + 1) - max(count) > k:
                count[ord(s[l]) - ord('A')] -= 1
                l += 1

            length = max(length, r - l + 1)
            r += 1
        return length

